query/expression.go

```diff
@@ -144,19 +145,52 @@ func (expr *metricFetchExpression) Evaluate(context EvaluationContext) (value, e
 }
 
 func (expr *functionExpression) Evaluate(context EvaluationContext) (value, error) {
-	switch expr.functionName {
-	case "+":
-		return evaluateBinaryOperation(context, expr.functionName, expr.arguments,
-			func(left, right float64) float64 { return left + right })
-	case "-":
-		return evaluateBinaryOperation(context, expr.functionName, expr.arguments,
-			func(left, right float64) float64 { return left - right })
-	case "/":
-		return evaluateBinaryOperation(context, expr.functionName, expr.arguments,
-			func(left, right float64) float64 { return left / right })
-	case "*":
-		return evaluateBinaryOperation(context, expr.functionName, expr.arguments,
-			func(left, right float64) float64 { return left * right })
+
+	name := expr.functionName
+	switch name {
+	case "+", "-", "*", "/":
```

Comment:
we should pull out `operatorMap` out, and have if statement on whether the value belongs on the `operatorMap` - this ensure that we don't have to manually keep `case` and `operatorMap` in sync.